src/EnergyPlus/CMakeLists.txt

```diff
@@ -746,10 +746,19 @@ SET( API_CORE_SRC
   api/datatransfer.cc
   api/EnergyPlusPgm.cc)
 
+option( BUILD_STATIC_ENERGYPLUS_API "Build a static library instead of a shared library for the EnergyPlus API" OFF )
+mark_as_advanced( BUILD_STATIC_ENERGYPLUS_API )
+if( BUILD_STATIC_ENERGYPLUS_API )
+	set(API_LIBRARY_TYPE STATIC)
+	ADD_DEFINITIONS(-DENERGYPLUS_STATIC_API)
+else()
+	set(API_LIBRARY_TYPE SHARED)
+endif()
```

Comment:
OK, so now if someone wants to develop tooling around E+ and wants to compile E+ statically, it's just a matter of setting this flag.